go.sum

```diff
@@ -457,6 +457,8 @@ github.com/stretchr/testify v1.9.0 h1:HtqpIVDClZ4nwg75+f6Lvsy/wHu+3BoSGCbBAcpTsT
 github.com/stretchr/testify v1.9.0/go.mod h1:r2ic/lqez/lEtzL7wO/rwa5dbSLXVDPFyf8C91i36aY=
 github.com/ugorji/go/codec v1.2.11 h1:BMaWp1Bb6fHwEtbplGBGJ498wD+LKlNSl25MjdZY4dU=
 github.com/ugorji/go/codec v1.2.11/go.mod h1:UNopzCgEMSXjBc6AOMqYvWC1ktqTAfzJZUZgYf6w6lg=
+github.com/valkey-io/valkey-go v1.0.48 h1:FSkZbS8X852icAWDfsBinQe0kUGO9+p/9Qj7bgRNHTw=
+github.com/valkey-io/valkey-go v1.0.48/go.mod h1:BXlVAPIL9rFQinSFM+N32JfWzfCaUAqBpZkc4vPY6fM=
```

Comment:
This is a fairly large architectural change. Looking at EDM-578, I don't see any detail on why a k/v store is needed or why this one was chosen.

Reply:
Right.  Added a link to the HLD doc in JIRA:
https://docs.google.com/document/d/1f_ZkgoNUtsU55AOyB_BjAZiLSDoaWF3rmS250VZLxmI/edit?usp=sharing